<img src= "../images/recommendation/agricultural_feed_banner.png" width = "800">
</img>


<h2 style= "color: #78C34D; text-align: left; font-weight: bold;">
Agricultural Feed Recommendation Engine
</h2>

<h3 style= "color: #CDAA2B; text-align: left; font-weight: bold;">
Business Understanding
</h3>

An agricultural supply company wants to recommend similar feed types to farmers based on feed performance data

The objective is to use the Chickwts dataset as a proxy for agricultural feed performance and build a recommendation system that identifies comparable feed types. PCA will be used for dimensionality reduction, while similarity metrics will be used to determine which feeds have similar performance characteristics

The final system will recommend feed types that may produce similar results

<h3 style= "color: #CDAA2B; text-align: left; font-weight: bold;">
1. Data Understanding
</h3>

In [1]:
# Import of required libs
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# load in the dataset
chick_df = pd.read_csv("../data/chickwts.csv")

# display first 5 rows
chick_df.head()

,weight,feed
0,179,horsebean
1,160,horsebean
2,136,horsebean
3,227,horsebean
4,217,horsebean


In [2]:
chick_df.shape

(71, 2)

In [3]:
chick_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   weight  71 non-null     int64 
 1   feed    71 non-null     object
dtypes: int64(1), object(1)
memory usage: 1.2+ KB


In [4]:
chick_df.isnull().sum()

weight    0
feed      0
dtype: int64

In [5]:
chick_df.duplicated().sum()

1

In [7]:
# investigate the duplicate
chick_df[chick_df.duplicated(keep=False)]

,weight,feed
24,248,soybean
35,248,soybean


In [6]:
chick_df["feed"].value_counts()

feed
soybean      14
linseed      12
sunflower    12
casein       12
meatmeal     11
horsebean    10
Name: count, dtype: int64

<h4 style= "color: #CDAA2B; text-align: left; font-weight: bold;">
Data Understanding Findings
</h4>

The dataset contains 71 observations and two variables: `weight`, representing chicken weight, and `feed`, representing the type of feed used.

The dataset contains no missing values. One pair of identical observations was identified, where two soybean-fed chickens have a recorded weight of 248.

Since the dataset does not contain a unique identifier, it cannot be determined whether this represents a duplicate record or two different chickens with the same weight and feed. Therefore, the observations were retained to avoid removing potentially valid data.

The dataset also contains six different feed types with relatively similar numbers of observations.

Since PCA cannot meaningfully operate directly on just weight and the categorical feed column

We'll first summarize each feed's performance:

In [8]:
feed_summary = chick_df.groupby("feed")["weight"].agg(
    mean_weight="mean",
    median_weight="median",
    std_weight="std",
    min_weight="min",
    max_weight="max"
).round(2)

feed_summary

,mean_weight,median_weight,std_weight,min_weight,max_weight
feed,,,,,
casein,323.58,342.0,64.43,216,404
horsebean,160.20,151.5,38.63,108,227
linseed,218.75,221.0,52.24,141,309
meatmeal,276.91,263.0,64.90,153,380
soybean,246.43,248.0,54.13,158,329
sunflower,328.92,328.0,48.84,226,423
